lti_consumer/tests/unit/test_lti_consumer.py

```diff
@@ -0,0 +1,664 @@
+"""
+Unit tests for LtiConsumerXBlock
+"""
+
+import unittest
+
+from datetime import datetime, timedelta
+from mock import Mock, PropertyMock, patch
+
+from lti_consumer.tests.unit.test_utils import FAKE_USER_ID, make_xblock, make_request
+
+from lti_consumer.lti_consumer import LtiConsumerXBlock, parse_handler_suffix
+from lti_consumer.exceptions import LtiError
+
+
+HTML_PROBLEM_PROGRESS = '<div class="problem-progress">'
+HTML_ERROR_MESSAGE = '<h3 class="error_message">'
+HTML_LAUNCH_MODAL_BUTTON = 'btn-lti-modal'
+HTML_LAUNCH_NEW_WINDOW_BUTTON = 'btn-lti-new-window'
+HTML_IFRAME = '<iframe'
+
+
+class TestLtiConsumerXBlock(unittest.TestCase):
+    """
+    Unit tests for LtiConsumerXBlock.max_score()
+    """
+
+    def setUp(self):
+        super(TestLtiConsumerXBlock, self).setUp()
+        self.xblock_attributes = {
+            'launch_url': 'http://www.example.com',
+        }
+        self.xblock = make_xblock('lti_consumer', LtiConsumerXBlock, self.xblock_attributes)
+
+
+class TestProperties(TestLtiConsumerXBlock):
+    """
+    Unit tests for LtiConsumerXBlock properties
+    """
+
+    def test_descriptor(self):
+        """
+        Test `descriptor` returns the XBLock object
+        """
+        self.assertEqual(self.xblock.descriptor, self.xblock)
+
+    def test_context_id(self):
+        """
+        Test `context_id` returns unicode course id
+        """
+        self.assertEqual(self.xblock.context_id, unicode(self.xblock.course_id))  # pylint: disable=no-member
+
+    def test_role(self):
+        """
+        Test `role` returns the correct LTI role string
+        """
+        self.xblock.runtime.get_user_role.return_value = 'student'
+        self.assertEqual(self.xblock.role, 'Student')
+
+        self.xblock.runtime.get_user_role.return_value = 'guest'
+        self.assertEqual(self.xblock.role, 'Student')
+
+        self.xblock.runtime.get_user_role.return_value = 'staff'
+        self.assertEqual(self.xblock.role, 'Administrator')
+
+        self.xblock.runtime.get_user_role.return_value = 'instructor'
+        self.assertEqual(self.xblock.role, 'Instructor')
+
+    def test_course(self):
+        """
+        Test `course` calls modulestore.get_course
+        """
+        mock_get_course = self.xblock.runtime.descriptor_runtime.modulestore.get_course
+        mock_get_course.return_value = None
```

Comment:
The runtime is a mock, since the unit tests are run without an actual XBlock runtime.  See https://github.com/edx/xblock-lti-consumer/pull/1/files#diff-b8f4cf5104f4cca5494e761e86fa3a0cR30.